lib/sidekiq/scheduler.rb

```diff
@@ -399,6 +404,24 @@ def set_schedule_state(name, state)
         Sidekiq.redis { |r| r.hset(schedules_state_key, name, JSON.generate(state)) }
       end
 
+      # Adds a Hash with schedule metadata as the last argument to call the worker.
+      # @example when parameter is an array
+      #   arguments_with_metadata(['arg1']) #=> ['arg1', {:scheduled_at => 1486039908.574729}]
+      # @example when parameter is a hash
+      #   arguments_with_metadata({arg1: 'value'}) #=> {:arg1 => 'value', :metadata => {:scheduled_at => 1486039908.574729}}
+      #
+      # @param args [Array|Hash]
+      # @return [Array|Hash] arguments with added metadata
+      def arguments_with_metadata(args)
+        metadata_hash = {scheduled_at: Time.now.to_f}
+        if args.is_a? Array
+          args.push(metadata_hash)
+        elsif args.is_a? Hash
+          args[:metadata] = metadata_hash
+        end
```

Comment:
When args is not an array (ie, `42`, `false`), the schedule metadata must be sent also